### PR TITLE
chore(react-core): add ui-react-core and configuration

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add initial `ui-react-core` package with `tsconfig`, `jest`, `tsup` and `eslint` configuration

#### Issue #, if available
NA

#### Description of how you validated changes
NA
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ErikCH — APPROVED** [review]
Looks like a good start. I know there has been talk about moving from `tsup`. But for now this is fine.

File: packages/react-core/.eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+module.exports = {
+  env: { node: true },
```
Comment (wlee221) [inline packages/react-core/.eslintrc.js:2]: QQ, why is `node` env set to `true` here?

File: packages/react-core/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "./package.json": "./package.json"
+  },
+  "types": "dist/index.d.ts",
+  "license": "Apache-2.0",
+  "files": [
+    "dist",
+    "LICENSE"
+  ],
+  "scripts": {
+    "prebuild": "rimraf dist",
```
Comment (wlee221) [inline packages/react-core/package.json:20]:
```suggestion
    "clean": "rimraf dist",
    "prebuild": "rimraf dist node_modules",
```
I'd add a `clean` here to be consistent with the other monorepo packages

**Review by wlee221 — APPROVED** [review]
LGTM, just non blocking comments

File: packages/react-core/.eslintrc.js (hunk 1 above)
Comment (calebpollman) [reply packages/react-core/.eslintrc.js:2]: It enables `node` global variables, more info [here](https://eslint.org/docs/user-guide/configuring/language-options#specifying-environments)

**Review by reesscot — APPROVED** [review]
LGTM, just a few comments

File: packages/react-core/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export * from './helloWorld';
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-core/src/index.ts:1]: Will delete all the example code before opening the PR against `main`

File: packages/react-core/package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "^7.26.0",
+    "eslint-plugin-react-hooks": "^4.3.0",
+    "jest": "^27.0.4",
+    "prettier": "2.4.1",
+    "react": "latest",
+    "react-test-renderer": "^17.0.2",
+    "rimraf": "^3.0.2",
+    "ts-jest": "^27.1.3",
+    "ts-morph": "^12.0.0",
```
[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-core/package.json:58]: Good catch, was copied over from `ui-react`. Will remove

File: packages/react-core/package.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@aws-amplify/ui-react-core",
+  "version": "0.1.0",
```
[the remark this answers is not in the document]
Comment (reesscot) [reply packages/react-core/package.json:3]: It's that helloWorld function @wlee221, it's a feature so it deserves a minor release 🤣

[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-core/package.json:3]: @reesscot coming in with the hot takes 😆 @wlee221 No particular reason, we used minor versioning when adding the `Notifications` category in the feature branch for `in-app-messaging` in AmplifyJS so i just defaulted to that as a pattern but am open to whatever is best practice for the UI team. Thoughts?

**Review by wlee221 — APPROVED** [review]

File: packages/react-core/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    "jsx": "preserve",
+    "importHelpers": true,
+    "lib": ["esnext", "dom", "dom.iterable", "scripthost"],
+    "module": "esnext",
+    "moduleResolution": "node",
+    "noImplicitAny": true,
+    "outDir": "dist",
+    "skipLibCheck": true,
+    "sourceMap": true,
+    "strict": false,
```
Comment (reesscot) [inline packages/react-core/tsconfig.json:22]: Let's turn this on now

**Pull request merged**